File: SiriusJavaClient-Web/src/main/java/org/sirius/client/web/classes/Frame.java
/**
 * 
 */
package org.sirius.client.web.classes;

import java.rmi.RemoteException;

import org.sirius.client.web.WebClient;
import org.sirius.client.web.core.Dimension;
import org.sirius.client.web.core.Point;

/**
 * @author Myk Kolisnyk
 *
 */
public class Frame {

	protected WebClient client = null;
	protected String locator = "";
	
	/**
	 * 
	 */
	public Frame(WebClient client,String locator) {
		this.client=client;
		this.locator=locator;
	}

	public boolean exists(){
		return false;
	}
	
	public String getURL() throws RemoteException{
		return this.client.core().getURL();
	}
	
	public String getLocator(){
		return this.locator;
	}
	
	public Point getLocation() throws RemoteException{
		return this.client.core().getLocation(null, locator);
	}
	
	public Dimension getSize() throws RemoteException {
		return this.client.core().getSize(null, locator);
	}
	
	public String innerHtml() throws RemoteException {
		return this.client.core().getAttribute(null, locator, "innerHTML");
	}
	
	public void switchTo(Frame frame) throws RemoteException{
		this.client.core().selectFrameByName(frame.getLocator());
	}
}

File: sirius-ruby-client-web/lib/sirius/web/classes/frame.rb
module Sirius
  module Client
    module Web
      module Classes
        class Frame
          attr_accessor :client
          attr_accessor :locator

          def initialize(client, locator)
            @client = client
            @locator = locator
          end

          def url
            @client.core.getURL
          end

          def location
            @client.core.getLocation(nil, locator)
          end

          def size
            @client.core.getSize(nil, locator);
          end

          def inner_html
            @client.core.getAttribute(nil, @locator, 'innerHTML')
          end

          def switch_to(frame)
            @client.core.selectFrameByName(frame.locator)
          end
        end
      end
    end
  end
end